views/clickhouse.go

```diff
@@ -125,17 +107,45 @@ func buildRealtimeViewsEventsQuery(spec QuerySpec) (string, []interface{}, error
 	return sql, args, nil
 }
 
-func filterMostRecent(res []RealtimeViewershipRow) []RealtimeViewershipRow {
-	if len(res) > 0 {
-		var filtered []RealtimeViewershipRow
-		mostRecentTimestamp := res[0].Timestamp
-		for _, r := range res {
-			if mostRecentTimestamp == r.Timestamp {
-				filtered = append(filtered, r)
-			} else {
-				return filtered
-			}
+// currentEventsQuery uses an unoptimized raw playback logs to favor latency over query speed.
+func currentEventsQuery(spec QuerySpec) squirrel.SelectBuilder {
+	return squirrel.Select(
+		"count(distinct session_id) as view_count").
+		From("viewership_events").
+		Where("user_id = ?", spec.Filter.UserID).
+		Where("server_timestamp > (toUnixTimestamp(now() - 30)) * 1000").
+		Limit(maxClickhouseResultRows)
+}
+
+// timeRangeEventsQuery uses an optimized materialized view "per minute" to favor query time against the latency.
+func timeRangeEventsQuery(spec QuerySpec) squirrel.SelectBuilder {
+	query := squirrel.Select(
+		"timestamp_ts",
+		"count(distinct session_id) as view_count",
+		"sum(buffer_ms) / sum(playtime_ms) as buffer_ratio",
```

Comment:
This is my fault. Need to update the denominator:
`sum(buffer_ms) / (sum(playtime_ms) + sum(buffer_ms))`

Reply:
Update in 206ac95872815f3ecb2d76bc3e1a151e38f7e08b